pb-test/gen/pb-jelly/proto_pbtest/src/pbtest3.rs.expected

```diff
@@ -4212,6 +4212,146 @@ impl ::pb_jelly::MessageDescriptor for TestMessage3RepeatedErrIfDefaultEnum {
   const FULL_NAME: &'static str = "pbtest.TestMessage3RepeatedErrIfDefaultEnum";
 }
 
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct TestMessage3NonOptionalBoxedMessage {
+  pub msg: ::std::boxed::Box<TestMessage3NonOptionalBoxedMessage_InnerMessage>,
+}
+impl ::std::default::Default for TestMessage3NonOptionalBoxedMessage {
+  fn default() -> Self {
+    TestMessage3NonOptionalBoxedMessage {
+      msg: ::std::default::Default::default(),
+    }
+  }
+}
+lazy_static! {
+  pub static ref TestMessage3NonOptionalBoxedMessage_default: TestMessage3NonOptionalBoxedMessage = TestMessage3NonOptionalBoxedMessage::default();
+}
+impl ::pb_jelly::Message for TestMessage3NonOptionalBoxedMessage {
+  fn compute_size(&self) -> usize  {
+    let mut size = 0;
+    let mut msg_size = 0;
+     {
+      let val = &*self.msg;
+      let l = ::pb_jelly::Message::compute_size(val);
+      msg_size += ::pb_jelly::wire_format::serialized_length(1);
+      msg_size += ::pb_jelly::varint::serialized_length(l as u64);
+      msg_size += l;
+    }
+    size += msg_size;
+    size
+  }
+  fn compute_grpc_slices_size(&self) -> usize  {
+    let mut size = 0;
+     {
+      let val = &*self.msg;
+      size += ::pb_jelly::Message::compute_grpc_slices_size(val);
+    }
+    size
+  }
+  fn serialize<W: ::pb_jelly::PbBufferWriter>(&self, w: &mut W) -> ::std::io::Result<()> {
+     {
+      let val = &*self.msg;
```

Comment:
This is the crux of the bug, previously we were not de-referencing the box, and we were trying to serialize a `Box<T>`. Now we de-reference it, and then serialize a `&T`